File: Python/8-kyu/11. Grasshopper - Summation/index.test.py
@test.describe('Basic tests')
def basic_tssts():
    test.assert_equals(summation(1), 1)
    test.assert_equals(summation(8), 36)
    test.assert_equals(summation(22), 253)
    test.assert_equals(summation(100), 5050)
    test.assert_equals(summation(213), 22791)

@test.describe('Random tests')
def random_tests():
    from random import randint
    
    solution = lambda num: num * (num + 1) / 2
    
    for i in range(100):
        rand = randint(1, 500)
        test.assert_equals(summation(rand), solution(rand))

File: JavaScript/8-kyu/11. Grasshopper - Summation/index.test.js
describe('summation', function () {
  it('should return the correct total', function () {
    Test.assertEquals(summation(1), 1)
    Test.assertEquals(summation(8), 36)
    Test.assertEquals(summation(22), 253)
    Test.assertEquals(summation(100), 5050)
    Test.assertEquals(summation(213), 22791)
  })
  it('random numbers test', function () {
    for (var i = 0; i < 100; i++) {
      var rand = Test.randomNumber() + 1
      Test.assertEquals(summation(rand), solution(rand))
    }
  })
  function solution (num) {
    var total = 0
    for (var i = 1; i <= num; i++) {
      total += i
    }
    return total
  }
})